src/view/gui/fMainWindow.cpp

```diff
@@ -1318,6 +1318,23 @@ void fMainWindow::SaveImage_withFile(int indexOfInputImageToWrite, QString saveF
 
       cbica::WriteImage< ImageTypeFloat3D >(infoChanger->GetOutput(), correctExtension(saveFileName_string));
     }
+    else if (cbica::ImageInfo(mSlicerManagers[index]->GetPathFileName()).GetImageDimensions() == 4)
+    {
+        // TODO: uncomment the below for saving 4D images
+        /*
+        auto reorientedImage4D = cbica::GetImageOrientation< ImageTypeFloat4D >(convertVtkToItk<ImageTypeFloat4D::PixelType, ImageTypeFloat4D::ImageDimension>(mSlicerManagers[index]->mImage), originalOrientation);
+        auto img = reorientedImage.second;
+        auto infoChanger = itk::ChangeInformationImageFilter< ImageType >::New();
+        infoChanger->SetInput(img);
+        infoChanger->ChangeDirectionOn();
+        infoChanger->ChangeOriginOn();
+        infoChanger->SetOutputDirection(originalDirection);
+        infoChanger->SetOutputOrigin(originalOrigin);
+        infoChanger->Update();
+
+        cbica::WriteImage< ImageTypeFloat4D >(infoChanger->GetOutput(), correctExtension(saveFileName_string));
+        */
+    }
```

Comment:
Originally I commented this out thinking I would need to write templated code for reorienting and saving the 4D image (because, AFAICT, our version of ITK doesn't have a working OrientImageFilter for 4D, causing all sorts of arcane template compiler  errors from ITK).

But since we don't actually reorient the 4D image itself to begin with (only the extracted 3D "slices" to compute mask size etc.), this isn't needed. So I uncommented the WriteImage and deleted the other lines.